src/controllers/bid.py

```diff
@@ -1,27 +1,131 @@
-from src.actions.game_actions import bid_on_power_plant, next_player
+from src.actions.game_actions import pass_current_bid, pass_initial_bid, set_current_bid, \
+    set_current_player, win_current_bid, next_phase, clear_bid, clear_bought_or_passed
+from src.actions.players_actions import remove_power_plant
+from src.decorators.connect import connect
 
 
-def bid(data, store):
+@connect
+def bid(bid_request, **kwargs):
     """ Handle how to update the game when a player tried to bid on a market item"""
-    player_id = data.get('player_id')
-    amount = data.get('amount')
-    card = data.get('card')
+    player_id = bid_request.get('player_id')
+    amount = bid_request.get('amount')
+    card = bid_request.get('card')
 
-    state = store['get_state']()
-    current_bid = state.get('game').get('current_bid')
+    get_state = kwargs['get_state']
+    dispatch = kwargs['dispatch']
+
+    initial_state = get_state()
+    if initial_state.get('game').get('phase') != 2:
+        return False
+    num_players = len(initial_state.get('players'))
+    current_bid = initial_state.get('game').get('current_bid')
+    # This indicates a passing bid
+    if amount == 0:
+        # If no one has bid, this indicates passing on an initial bid
+        if not current_bid:
```

Comment:
Isn't current bid initialized to be a `dict`, so it won't be empty to start? Or did we change that.

Reply:
Should check that there is no card being bid on